imx/deposit.go

```diff
@@ -37,9 +37,9 @@ type ERC721Deposit struct {
 }
 
 // NewETHDeposit instantiates a new ETHDeposit object with the given amount.
-func NewETHDeposit(amount string) *ETHDeposit {
+func NewETHDeposit(amountInWei string) *ETHDeposit {
 	this := ETHDeposit{}
-	this.Amount = amount
+	this.Amount = amountInWei
```

Comment:
I like the clarity here :+1: 

However, a more type-safe way is to introduce unit types, for example:

```go
type Wei = uint64
const (
    wei = Wei(1)
    gwei = Wei(1000000000)
    eth = Wei(1000000000000000000)
)
```

and your function should take the type `Wei` instead of string:

```
func NewETHDeposit(amount Wei) *ETHDeposit {
    //...
    this.Amount = strconv.FormatUint(uint64(amount), 10)
}
```

Reply:
This is similar to how time works in Go:
https://linuxhint.com/golang-time-duration/

Reply:
the caller would use your function like so:

```go
depo := NewETHDeposit(100 * wei);

// or 
depo := NewETHDeposit(2 * Eth);
```